keyboards/keychron/q1/q1.c

```diff
@@ -0,0 +1,165 @@
+/* Copyright 2021 @ Keychron (https://www.keychron.com)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "q1.h"
+
+
+void set_target_os(bool layer)
+{
+	if(layer)
+	{
+		layer_off(2);
+	}
+	else
+	{
+		layer_on(2);
```

Comment:
While this is okay, I would prefer that this use the default layer functionality, as it's designed more for this. 

However, this would require re-ordering your layers.  

Specifically `default_layer_set`, from:
https://docs.qmk.fm/#/feature_layers

Reply:
I agree with you that use `default_layer_set` rather than `layer_on/layer_off`, but I don't understand why I need to re-order the layers, could you elaborate?  I replaced  `layer_on/layer_off` to 'default_layer_set` and made a test, everything works fine.

Reply:
the reason for the re-order is that anything below the default layer becomes inaccessible.

If that's okay, then you don't need to re-order.